### ISSUE TITLE
[clang-tidy] bugprone-unused-return-value false positive will matching assignment

### ISSUE THREAD

**Issue by HerrCai0907** [llvm/llvm-project#84480]
It seems like this is a bigger issue.

If you use `::.*` for `CheckedFunctions` to match all functions it will report the warning on *every* assignment (`=` and `+=`) regardless of type.

_Originally posted by @firewave in https://github.com/llvm/llvm-project/issues/84314#issuecomment-1984026830_
            

Here's an example:

```cpp
#include <string>

class C {
public:
    void setS(const char* s) {
        mS = s;
    }
private:
    std::string mS;
};
```

```
test.cpp:6:9: error: the value returned by this function should not be disregarded; neglecting it may lead to errors [bugprone-unused-return-value,-warnings-as-errors]
    6 |         mSt = s;
      |         ^~~~~~~
```

(No godbolt link as I could not figure out how to make the passing of the option work)

_Originally posted by @firewave in https://github.com/llvm/llvm-project/issues/84314#issuecomment-1984118735_
